### PR TITLE
Use bazel workspace status to generate versions

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: taskcluster/tc-tests-utils.sh [hunk 1]
```diff
@@ -811,7 +811,7 @@ do_bazel_build()
   fi;
 
   bazel ${BAZEL_OUTPUT_USER_ROOT} build \
-    -s --explain bazel_monolithic.log --verbose_explanations --experimental_strict_action_env --config=monolithic -c opt ${BAZEL_BUILD_FLAGS} ${BAZEL_TARGETS}
+    -s --explain bazel_monolithic.log --verbose_explanations --experimental_strict_action_env --workspace_status_command="bash native_client/bazel_workspace_status_cmd.sh" --config=monolithic -c opt ${BAZEL_BUILD_FLAGS} ${BAZEL_TARGETS}
```
Comment (lissyx) [inline taskcluster/tc-tests-utils.sh]: So, no spurious rebuilds with that ?

File: taskcluster/tc-tests-utils.sh [hunk 2]
```diff
@@ -826,14 +826,6 @@ shutdown_bazel()
   bazel ${BAZEL_OUTPUT_USER_ROOT} shutdown
 }
 
-do_bazel_shared_build()
```
Comment (lissyx) [inline taskcluster/tc-tests-utils.sh]: Nice removal, I should try and find some time to fix / refactor this :)

File: native_client/bazel_workspace_status_cmd.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#
+# If the script exits with non-zero code, it's considered as a failure
+# and the output will be discarded.
+
+# The code below presents an implementation that works for git repository
+tf_git_rev=$(git describe --long --tags)
+echo "STABLE_TF_GIT_VERSION ${tf_git_rev}"
+
+pushd native_client
+ds_git_rev=$(git describe --long --tags)
```
Comment (lissyx) [inline native_client/bazel_workspace_status_cmd.sh]: Does that works with `git submodules` as well?

File: native_client/dotnet/README.md [hunk 4]
```diff
@@ -107,14 +107,14 @@ At this point we are ready to start building the `native_client`, go to `tensorf
 We will add AVX/AVX2 support in the command, please make sure that your CPU supports these instructions before adding the flags, if not you can remove them.
 
 ```bash
-bazel build -c opt --copt=/arch:AVX --copt=/arch:AVX2 //native_client:libdeepspeech.so
+bazel build --workspace_status_command="bash native_client/bazel_workspace_status_cmd.sh" -c opt --copt=/arch:AVX --copt=/arch:AVX2 //native_client:libdeepspeech.so
```
Comment (lissyx) [inline native_client/dotnet/README.md]: Oh, looks like I missed it, you did update it

File: taskcluster/tc-tests-utils.sh (hunk 1 above)
Comment (reuben) [reply taskcluster/tc-tests-utils.sh]: Nope. All the spurious rebuilds tests that were failing before are now green.

File: native_client/bazel_workspace_status_cmd.sh (hunk 3 above)
Comment (reuben) [reply native_client/bazel_workspace_status_cmd.sh]: Yep!

**Pull request merged**